### PR TITLE
[prompt] update prompt api & add prefix template

### PR DESCRIPTION
**Pull request by LemonNoel**
### PR types

New features

### PR changes

APIs

### Description

1. Update Template String with 2 keywords and 7 attributes.
2. Add `PrefixTemplate`.
3. Optimize the implementation of `Verbalizer`.
4. Delete `MultiMaskVerbalizer`.
5. Delete `InputExample` and `InputFeatures` and use `dict` instead for flexibility, where keyword `labels` denotes the label id of a data sample.
6. Update APIs in text classification applications.

**Template new features**
---
**Keywords**
- `options`: Used list of labels in template by defining label_file_path / option_list_name_in_example_dict.
- `prefix`: Add prefix soft tokens to inputs on every layer.

**Attributes**
- `position`: Common. Define the start position id of following blocks.
- `token_type`: Common. Define the token type id of following blocks.
- `add_omask`: For `options`. Add `[O-MASK]` to each option.
- `add_prompt`: For `options`. Add prompt text to each option.
- `encoder`: For `soft` and `prefix`. Define the encoder type.
- `hidden_size`: For `soft` and `prefix`. Define the hidden_size of encoder.
- `length`: For `mask`, `soft` and `prefix`. Define the length of tokens.

### CODE REVIEW

File: applications/text_classification/hierarchical/few-shot/README.md [hunk 1]
```diff
@@ -65,9 +65,9 @@
 
    内存: 630 GB
 
-3. PaddlePaddle 版本：2.3.1
+3. PaddlePaddle 版本：2.4rc
 
-4. PaddleNLP 版本：2.3.5 (develop)
+4. PaddleNLP 版本：2.4.2 (develop)
```
Comment (wawltor) [inline applications/text_classification/hierarchical/few-shot/README.md]: 这个要么直接2.4.3版本？ 后面也可以直接2.4.3版本

最好能依赖关系放在requirements.txt里面

File: applications/text_classification/hierarchical/few-shot/README.md [hunk 2]
```diff
@@ -352,9 +353,9 @@ python infer.py --model_path_prefix checkpoints/export/model --data_dir ./data -
 可配置参数说明：
 
 - `model_path_prefix`: 导出的静态图模型路径及文件前缀。
-- `model_name_or_path`: 内置预训练模型名，或者模型参数配置目录路径，用于加载tokenizer。默认为`ernie-3.0-base-zh`。
+- `model_name`: 内置预训练模型名，用于加载tokenizer。默认为`ernie-3.0-base-zh`。
```
Comment (wawltor) [inline applications/text_classification/hierarchical/few-shot/README.md:356]: 这里为什么model_name_or_path-> model_name ，不可以直接传入一个path进去？

Comment (LemonNoel) [reply applications/text_classification/hierarchical/few-shot/README.md:356]: 这里`model_name`只是为了加载预训练模型相关的参数，用于数据预处理。上边`model_path_prefix`才是 Prompt 模型的参数地址，之前两个都带 `path` 容易混淆。

File: applications/text_classification/hierarchical/few-shot/README.md (hunk 1 above)
Comment (LemonNoel) [reply applications/text_classification/hierarchical/few-shot/README.md]: 已修改为2.4.3版本。相关依赖放在了 `requirements_cpu.txt` 和 `requirements_gpu.txt` 文件。

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**